configGlobal_test.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/spf13/viper"
+)
+
+func TestValidConfigurationChannel(t *testing.T) {
+	viper.Reset()
+	viper.SetConfigFile("test/assets/globalConfig.yml")
```

Comment:
The problem I faced was that viper is being used in multiple places. And I believe I should have actually mocked viper, but I have no experience in mocking in go and didn't want to add extra dependencies (i.e. mocking library) without discussing it first. So I tried to write unit tests which are more encapsulated and focusses on single parts instead of the main entry point. But to be honest writing good test is something I've always struggled with, so it's good your calling me on it ;).

I'll give it another shot.

Reply:
I think one way to handle this: loading the global configuration is only done once in the normal code path. So be sure the code completely resets the viper object used for global configuration at the beginning of loading the global configuration. That way, you can call it multiple times in unit tests with no harm.

Mocking is important for complete subsystems (i.e. a database, an authentication server, etc). While you could mock viper, I think that's not really necessary in this case (viper is pretty light weight). I think there are two key points for a unit test:

1. Test all possible aspects of the code as possible (always strive for as close to 100% test coverage for the tested code that you can get), and
2. Actually test the code you're trying to test.

In the case of `globalConfig_test.go`, I would expect that to test major sections of global configuration, and it's not doing that today. It's a good start, but it should be more complete.